### PR TITLE
3131 errror in calculation of temperature margin bug

### PR DESCRIPTION
**Pull request by mkovari**
## Description

I have replaced the home-made Newton-Raphson solver and the secant solver with `newton` for calculating the temperature margin of superconductors.

## Checklist

I confirm that I have completed the following checks:

- [X] No large differences in the regression tests except for constraint residues.
- [X] No new tests
- [X] No doc changes

### CODE REVIEW

**Comment by jonmaddock** [comment]
Hi Michael. Can you resolve the conflicts on this branch, then possibly tidy up the commit history with an interactive rebase if you feel that would make these changes clearer?

**Comment by mkovari** [comment]
Jon,
I have had a look at the conflict but I don’t I understand what has caused it.  I have attempted to resolve it by simply restoring my code, but now the tests are failing when they weren’t yesterday.  I am afraid I don’t know what’s going on.

All the best,
Michael

**Comment by mkovari** [comment]
Anyway - there are now no merge conflicts.
I don't know what an interactive rebase is.

**Comment by mkovari** [comment]
I believe this is now ready to merge.   I did both the things we discussed:
1. Replaced `brentq` by `newton`. 
2. Restored the commented-out integration tests, changed the  value of `rohc` to something more sensible, and changed the two assertions that failed as a result.

File: tests/integration/data/test_solver.conf [hunk 1]
```diff
@@ -2,7 +2,8 @@
 *
 
 * No iterations
-NITER = 10
+# NITER = 10
+NITER = 1
```
Comment (jonmaddock) [inline tests/integration/data/test_solver.conf:6]: What's the purpose of this change? The rest of the commit appears to be to do with getting the sctfcoil and superconductors tests to pass.

File: tests/integration/test_pfcoil_int.py [hunk 2]
```diff
@@ -35,241 +37,241 @@ def pfcoil():
...
+#     monkeypatch.setattr(tfv, "b_crit_upper_nbti", 1.486e1)
+#     monkeypatch.setattr(tfv, "b_crit_upper_nbti", 9.04)
+
+#     pfcoil.ohcalc()
+
+#     assert pytest.approx(pfv.bpf[4]) == 9.299805e2
+#     assert pytest.approx(pfv.rjohc) == -7.728453e9
+
+
+def test_efc(pfcoil: PFCoil[CsFatigue], monkeypatch: pytest.MonkeyPatch):
```
Comment (jonmaddock) [inline tests/integration/test_pfcoil_int.py]: `pfcoil: PFCoil[CsFatigue]`: I'm not sure that's the correct type hint.

File: tests/integration/test_pfcoil_int.py [hunk 3]
```diff
@@ -35,241 +37,241 @@ def pfcoil():
...
-    monkeypatch.setattr(tfv, "b_crit_upper_nbti", 9.04)
-
-    pfcoil.ohcalc()
-
-    assert pytest.approx(pfv.bpf[4]) == 9.299805e2
-    assert pytest.approx(pfv.rjohc) == -7.728453e9
-
-
-def test_efc(pfcoil, monkeypatch):
+# def test_pfcoil(monkeypatch, pfcoil):
```
Comment (jonmaddock) [inline tests/integration/test_pfcoil_int.py]: As all this is commented back in later, can this be fixed up to reduce churn in this PR?

File: tests/integration/test_pfcoil_int.py [hunk 4]
```diff
@@ -271,7 +271,7 @@ def pfcoil():
 #     assert pytest.approx(pfv.rjohc) == -7.728453e9
 
 
-def test_efc(pfcoil: PFCoil[CsFatigue], monkeypatch: pytest.MonkeyPatch):
+def test_efc(pfcoil: PFCoil, monkeypatch: pytest.MonkeyPatch):
```
Comment (jonmaddock) [inline tests/integration/test_pfcoil_int.py:274]: As this corrects an earlier error, can the previous commit be fixed up with this to reduce churn please?

File: process/pfcoil.py [hunk 5]
```diff
@@ -2889,26 +2889,27 @@ def superconpf(
...
             if isumat == 3:
-                args = (isumat, jsc, bmax, strain, bc20m, tc0m, c0)
+                arguments = (isumat, jsc, bmax, strain, bc20m, tc0m, c0)
             else:
-                args = (isumat, jsc, bmax, strain, bc20m, tc0m)
+                arguments = (isumat, jsc, bmax, strain, bc20m, tc0m)
 
-            t_zero_margin, root_result = optimize.brentq(
+            another_estimate = 2 * thelium
+            t_zero_margin, root_result = optimize.newton(
```
Comment (jonmaddock) [inline process/pfcoil.py:2899]: As the solver has been changed from newton-raphson to brentq to newton, can this churn be fixed up please?

File: tests/integration/test_pfcoil_int.py [hunk 6]
```diff
@@ -37,238 +37,238 @@ def pfcoil():
...
-#     monkeypatch.setattr(tfv, "tmargmin_cs", 1.5)
-#     monkeypatch.setattr(tfv, "temp_margin", 0.0)
-#     monkeypatch.setattr(tfv, "b_crit_upper_nbti", 1.486e1)
-#     monkeypatch.setattr(tfv, "b_crit_upper_nbti", 9.04)
-
-#     pfcoil.ohcalc()
-
-#     assert pytest.approx(pfv.bpf[4]) == 9.299805e2
-#     assert pytest.approx(pfv.rjohc) == -7.728453e9
+def test_pfcoil(monkeypatch, pfcoil):
```
Comment (jonmaddock) [inline tests/integration/test_pfcoil_int.py]: Commenting back in of commented-out function: please reduce churn.

File: process/pfcoil.py [hunk 7]
```diff
@@ -2909,8 +2909,7 @@ def superconpf(
                 full_output=True,
                 disp=False,
             )
-            print(root_result)  # Diagnostic for newton method
-            print("bmax = ", bmax)
+            # print(root_result)  # Diagnostic for newton method
```
Comment (jonmaddock) [inline process/pfcoil.py:2912]: Could be fixed-up.

File: tests/integration/data/test_solver.conf (hunk 1 above)
Comment (mkovari) [reply tests/integration/data/test_solver.conf:6]: Reverted

File: process/pfcoil.py (hunk 7 above)
Comment (mkovari) [reply process/pfcoil.py:2912]: Comment line removed

File: process/pfcoil.py (hunk 5 above)
Comment (mkovari) [reply process/pfcoil.py:2899]: Will do using rebase

File: tests/integration/test_pfcoil_int.py (hunk 6 above)
Comment (mkovari) [reply tests/integration/test_pfcoil_int.py]: Will do using rebase

File: tests/integration/test_pfcoil_int.py (hunk 4 above)
Comment (mkovari) [reply tests/integration/test_pfcoil_int.py:274]: Will do using rebase

File: process/maths_library.py [hunk 8]
```diff
@@ -4,66 +4,66 @@
 Fortran; this is a temporary measure during the Python conversion process.
 """
 
-import numpy as np
+# import numpy as np
 
 
-def secant_solve(f, x1, x2, opt_tol=None):
-    """Solve an equation f(x)=0.
+# def secant_solve(f, x1, x2, opt_tol=None):
```
[the remark this answers is not in the document]
Comment (mkovari) [reply process/maths_library.py:10]: Deleted

**Comment by mkovari** [comment]
I think the conflict is resolved now.

**Review by jonmaddock — APPROVED** [review]

**Comment by jonmaddock** [comment]
I'm afraid there are still conflicts that need to be resolved. Please resolve them by rebasing, then re-request a review. Thanks.

**Comment by mkovari** [comment]
GitHub says this branch has no conflicts with the base branch.  What conflicts are you thinking of?
